include/spirv-tools/libspirv.h

```diff
@@ -425,9 +425,70 @@ typedef enum spv_binary_to_text_options_t {
   // Reorder blocks to match the structured control flow of SPIR-V to increase
   // readability.
   SPV_BINARY_TO_TEXT_OPTION_REORDER_BLOCKS = SPV_BIT(9),
+  // Add style annotations to the disassembly, which can be post-processed by
+  // the caller as needed. The style markers begin with character 0x0E ('\e')
+  // and end in 0x0F ('\f'). In between, there are 0 or more characters that
+  // indicate the style, defined by spv_binary_to_text_style.
+  //
+  // For example:
+  //
+  //   \efp\f: Begin styling a float-y pointer type
+  //   \ec\f: Begin styling a constant value (of a non-numeric type)
+  //   \e\f: End of style
+  //
+  // The application could for instance turn `\ehp\f%16\e\f` into a yellow
+  // underlined %16, while `\ebc\f%9\e\f` is an italic cyan %9.
+  //
+  // This option overrides SPV_BINARY_TO_TEXT_OPTION_COLOR.
+  SPV_BINARY_TO_TEXT_OPTION_STYLE = SPV_BIT(10),
   SPV_FORCE_32_BIT_ENUM(spv_binary_to_text_options_t)
 } spv_binary_to_text_options_t;
 
+typedef enum spv_binary_to_text_style_t {
```

Comment:
This seems very limiting.  It's all about certain aspects of types.  That seems limiting. Other uses might want to classify pointers by address space/storage class, for example.   There's a lot of other static info you might want to see about a SPIR-V module.

Using a single letter to encode stuff, then assume you can arbitrarily combine them, seems likely to lead to confusion/errors.